File: DataStructure/SimulateStackWithQueue/MyStack.java
class MyStack {
    ArrayList<Integer> qMaster = new ArrayList<Integer>();
    ArrayList<Integer> qTmp = new ArrayList<Integer>();
    public void push(int x) {
        qTmp.add(Integer.valueOf(x));
        while (qMaster.size() >0) {
            qTmp.add(qMaster.remove(0));
        }
        ArrayList<Integer> tmp = qTmp;
        qTmp = qMaster;
        qMaster = tmp;
        return;
    }

    // Removes the element on top of the stack.
    public void pop() {
        if (qMaster.size() >0) {
          qMaster.remove(0);
        }
        return ;
    }

    // Get the top element.
    public int top() {
        if (qMaster.size() >0) {
          return qMaster.get(0);
        }
        return 0;
    }

    // Return whether the stack is empty.
    public boolean empty() {
        return qMaster.size()==0;
    }
}

File: Stack/[Leetcode] 225. Implement Stack using Queues/test225-go/lzhou.programmingtest.leetcode/question225/mystack.go
package question225

import "container/list"

type MyQueue struct {
	data *list.List
}

func NewQueue() *MyQueue {
	q := new(MyQueue)
	q.data = list.New()
	return q
}

func (this *MyQueue) Len() int {
	return this.data.Len()
}

func (this *MyQueue) Offer(val interface{}) {
	this.data.PushBack(val)
}

func (this *MyQueue) Peek() interface{} {
	if 0 == this.data.Len() {
		return nil
	}
	if v := this.data.Front(); v == nil {
		return nil
	} else {
		return v.Value
	}
}

func (this *MyQueue) Poll() interface{} {
	if 0 == this.data.Len() {
		return nil
	}
	if v := this.data.Front(); v == nil {
		return nil
	} else {
		return this.data.Remove(v)
	}
}

/**********************/

type MyStack struct {
	queue1 *MyQueue
	queue2 *MyQueue
}

/** Initialize your data structure here. */
func Constructor() MyStack {
	var ret MyStack
	ret.queue1 = NewQueue()
	ret.queue2 = NewQueue()
	return ret
}

/** Push element x onto stack. */
func (this *MyStack) Push(x int) {
	this.queue2.Offer(x)
	for this.queue1.Len() > 0 {
		this.queue2.Offer(this.queue1.Poll())
	}
	this.queue1, this.queue2 = this.queue2, this.queue1
}

/** Removes the element on top of the stack and returns that element. */
func (this *MyStack) Pop() int {
	v, ok := this.queue1.Poll().(int)
	if ok {
		return v
	} else {
		return 0
	}
}

/** Get the top element. */
func (this *MyStack) Top() int {
	v, ok := this.queue1.Peek().(int)
	if ok {
		return v
	} else {
		return 0
	}
}

/** Returns whether the stack is empty. */
func (this *MyStack) Empty() bool {
	return this.queue1.Len() == 0
}

/**
 * Your MyStack object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(x);
 * param_2 := obj.Pop();
 * param_3 := obj.Top();
 * param_4 := obj.Empty();
 */
